src/pudl/output/sec10k.py

```diff
@@ -106,3 +110,135 @@ def out_sec10k__changelog_company_name(
         name_changelog["company_name_old"] != name_changelog["company_name_new"]
     ]
     return name_changelog
+
+
+def _get_n_duplicate_filer_records(company_info_df, ownership_df, key):
+    filers_count_df = (
+        company_info_df.groupby(key).size().reset_index(name="filer_count")
+    )
+    non_unique_filers_df = filers_count_df[filers_count_df.filer_count > 1]
+    return len(
+        ownership_df.merge(
+            non_unique_filers_df,
+            on=key,
+            how="inner",
+        ).drop_duplicates(subset=key)
+    )
+
+
+@dg.asset(
+    io_manager_key="pudl_io_manager",
+    group_name="out_sec10k",
+)
+def out_sec10k__parents_and_subsidiaries(
+    core_sec10k__quarterly_exhibit_21_company_ownership: pd.DataFrame,
+    out_sec10k__quarterly_company_information: pd.DataFrame,
+    core_sec10k__assn_exhibit_21_subsidiaries_and_filers: pd.DataFrame,
+    core_sec10k__assn_exhibit_21_subsidiaries_and_eia_utilities: pd.DataFrame,
+    core_eia__entity_utilities: pd.DataFrame,
+) -> pd.DataFrame:
+    """Denormalized output table with Sec10k company attributes and ownership info linked to EIA."""
+    # We want to merge company info records onto ownership records on filename, report_date, and CIK
+    # but this key is not unique in the company info records because it doesn't include filer_count.
+    # Log the number of non unique records from the information table that appear as
+    # parent companies in the ownership table.
+    parent_co_merge_key = [
+        "filename_sec10k",
+        "parent_company_central_index_key",
+        "report_date",
+    ]
+    n_parent_dupes = _get_n_duplicate_filer_records(
+        out_sec10k__quarterly_company_information.rename(
+            columns={"central_index_key": "parent_company_central_index_key"}
+        ),
+        core_sec10k__quarterly_exhibit_21_company_ownership,
+        key=parent_co_merge_key,
+    )
+    if n_parent_dupes > 0:
+        logger.warning(
+            f"""There are {n_parent_dupes} duplicate filenames, CIKs, and report date records that are parent companies in the Ex. 21 ownership table."""
+        )
+    # merge parent attributes on
+    parents_info_df = out_sec10k__quarterly_company_information.add_prefix(
+        "parent_company_"
+    ).drop(columns=["parent_company_company_name"])
+    df = core_sec10k__quarterly_exhibit_21_company_ownership.merge(
+        parents_info_df,
+        how="left",
+        left_on=["filename_sec10k", "parent_company_central_index_key", "report_date"],
+        right_on=[
+            "parent_company_filename_sec10k",
+            "parent_company_central_index_key",
+            "parent_company_report_date",
```

Comment:
See my comments in the SEC 10-K resource definitions about what the meaning of these columns is in the context of this table.